src/framework/components/camera/post-effect-queue.js

```diff
@@ -404,6 +380,8 @@ class PostEffectQueue {
 
         for (let i = 0, len = effects.length; i < len; i++) {
             const fx = effects[i];
+            if (fx.effect.resize !== undefined)
+                fx.effect.resize();
```

Comment:
should this be called after _resizeOffscreenTarget so that the effect can use the siez of those targets for resizing?

Reply:
Is it better that they depend on the input target or the graphics device size? The input target could be quarter res but the depth target full res, for example.

Reply:
what if we render with camera to a different size or even different aspect than device size? Device size should not be used at all.

Reply:
True. Okay I'll change this to be dependent on the input target.

Reply:
Okay, so looking back at this, every post effect is created without the input target and rely on the device size on creation. I think considering these parameters, the base size of the effect shouldn't be handled by the effect at all, but rather by the post effects queue.

So on init, we should also pass a context containing the size for the effect. I guess we can add that to the list of things we want to do when refactoring the post effects stuff 🙂 .